File: Algorithm/Climbing_Staircase/climbing_staircase.py
def climbing_staircase(n):
    num_of_ways = [1,2]
    for i in range(2, n):
        num_of_ways.append(num_of_ways[i - 1] + num_of_ways[i-2])
    return num_of_ways[n -1]

print(climbing_staircase(5))

File: Algorithm/Climbing_Staircase/climbing_staircase.js

const climbing_staircase = (n) => {
    const noOfWays = [ 1, 2 ];
    for (i = 2; i <= n; i++) {
        noOfWays[ i ] = noOfWays[ i - 1 ] + noOfWays[ i - 2 ];
    }
    return noOfWays[ n - 1 ];
};

console.log(climbing_staircase(4));
// Big-O = O(n)